libcxx/include/optional

```diff
@@ -482,6 +474,16 @@ struct __optional_storage_base<_Tp, true> {
         __construct(std::forward<_That>(__opt).__get());
     }
   }
+
+  template <class _Up>
+  _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 void __assign_from_val(_Up&& __val) {
+    __value_ = std::addressof(__val);
+  }
+
+  _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 void __swap(__optional_storage_base& __rhs) {
+    using std::swap;
+    swap(__value_, __rhs.__value_);
```

Comment:
Why not `std::swap`?

Reply:
Ah. This is specified by the current standard wording - [[optional.ref.swap]/1](https://eel.is/c++draft/optional.ref.swap#1) that uses unqualified `swap`. But I think we should unconditionally use `std::swap`. I'm to submit an LWG issue.

Let's make this function `noexcept` and unconditionally call `std::swap`.